lib/cheerio.js

```diff
@@ -53,7 +53,7 @@ var Cheerio = (module.exports = function (selector, context, root, options) {
 
   if (root) {
     if (typeof root === 'string') root = parse(root, this.options, false);
-    this._root = Cheerio.call(this, root);
+    this._root = new Cheerio(root);
```

Comment:
Finally figured out what's wrong here: This should be `new this.constructor(root)`. Incorporating this in a different PR.